### PR TITLE
Add support for Debian 12

### PR DESCRIPTION
**Pull request by cFire**
#### Pull Request (PR) description
Created the template for Debian 12 based on the default jail.conf that ships with the fail2ban bookworm package.

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

**Comment by cFire** [comment]
Are Debian 12 tests something I can add in the PR? Or do they need to be configured elsewhere?

File: templates/Debian/12/etc/fail2ban/jail.conf.epp [hunk 1]
```diff
@@ -0,0 +1,1080 @@
...
+
+[scanlogd]
+enabled   = <%= "scanlogd" in $fail2ban::jails %>
+logpath   = %(syslog_local0)s
+banaction = %(banaction_allports)s
+
+[monitorix]
+enabled = <%= "monitorix" in $fail2ban::jails %>
+port    = 8080
+logpath = /var/log/monitorix-httpd
```
Comment (kenyon) [inline templates/Debian/12/etc/fail2ban/jail.conf.epp:1080]: Add final newline:

```suggestion
logpath = /var/log/monitorix-httpd

```

**Review by kenyon — CHANGES_REQUESTED** [review]
Tests for Debian 12 bookworm won't work until packages are available: https://tickets.puppetlabs.com/browse/PA-4995

**Comment by tskirvin** [comment]
This doesn't work on Debian 12.  The default `fail2ban::action` is still set to `action_mb`, which doesn't exist for Debian 12 in this template. FYI.

**Comment by cFire** [comment]
> This doesn't work on Debian 12. The default `fail2ban::action` is still set to `action_mb`, which doesn't exist for Debian 12 in this template. FYI.

Thanks for letting me know. I never noticed it was missing since I set a different action in my config. I've merged the actions section from the Debian 11 config into the current file but I don't have a good way to test the `action_mb` on my system, so if you have some way to test and verify that would be great!

**Comment by tskirvin** [comment]
> > This doesn't work on Debian 12. The default `fail2ban::action` is still set to `action_mb`, which doesn't exist for Debian 12 in this template. FYI.
> 
> Thanks for letting me know. I never noticed it was missing since I set a different action in my config. I've merged the actions section from the Debian 11 config into the current file but I don't have a good way to test the `action_mb` on my system, so if you have some way to test and verify that would be great!

I did a tiny amount of checking, and it works now, but honestly I was convinced to switch back to something besides action_mb anyway because I didn't actually want the emails.  Thanks for the encouragement!

**Comment by kwisatz** [comment]
This seems to have been approved. Will we see a new release with this included anytime soon?

**Review by Valantin — APPROVED** [review]

**Pull request merged**